packages/cubejs-teradata-jdbc-driver/src/TeradataQuery.ts

```diff
@@ -0,0 +1,139 @@
+    const [number, type] = this.parseInterval(interval);
+    console.log(`(${date} + INTERVAL '${number}' ${type})`);
+    return `(${date} + INTERVAL '${number}' ${type})`;
+  }
+
+  public timeGroupedColumn(granularity: string, dimension: string): string {
+    console.log(`TRUNC(CAST(${dimension} AS TIMESTAMP(6)), '${GRANULARITY_TO_INTERVAL[granularity]}')`);
+    return `TRUNC(CAST(${dimension} AS TIMESTAMP(6)), '${GRANULARITY_TO_INTERVAL[granularity]}')`;
+  }
+
+  public rowNumberColumn() {
+    return ', ROW_NUMBER() OVER (ORDER BY 1) AS RowNum_ ';
+  }
+
+  public groupByDimensionLimit() {
+    let startWindow = 0;
+    let endWindow = 1000;
+    if (this.offset && parseInt(this.offset, 10)) {
+      startWindow = this.rowLimit === null ? 0 : (this.rowLimit && parseInt(this.rowLimit, 10) || 0);
+      endWindow = startWindow + parseInt(this.offset, 10);
+    } else {
+      endWindow = this.rowLimit === null ? 0 : (this.rowLimit && parseInt(this.rowLimit, 10) || 1000);
+    }
+
+    return ` QUALIFY RowNum_ BETWEEN ${startWindow} AND ${endWindow}`;
+  }
+
+  public commonQuery() {
+    return `
+      SELECT
+      ${this.baseSelect()}
+      ${this.rowNumberColumn()}
+      FROM
+      ${this.query()}
+      ${this.groupByClause()}
+      ${this.baseHaving(this.measureFilters)}
+      ${this.groupByDimensionLimit()}`;
+  }
+
+  public simpleQuery() {
+    // eslint-disable-next-line prefer-template
+    const inlineWhereConditions: any[] = [];
+    const commonQuery = this.rewriteInlineWhere(() => this.commonQuery(), inlineWhereConditions);
+    const aliases = this.baseSelect()
+      .split(' ')
+      .map(word => word.replace(',', ''))
+      .filter((word) => word.includes('__'))
+      .toString();
+    // eslint-disable-next-line prefer-template
+    return `
+        WITH WINDOW_TABLE AS  (
+        ${commonQuery} 
+        ${this.baseWhere(this.allFilters.concat(inlineWhereConditions))}`
+       + `) SELECT ${aliases} FROM WINDOW_TABLE`
+       + this.orderBy();
+  }
+  
+  public escapeColumnName(name: string) {
+    return `"${name}"`;
+  }
+
+  public renderDimensionCaseLabel(label: any, cubeName: any) {
+    if (typeof label === 'object' && label.sql) {
+      return this.evaluateSql(cubeName, label.sql, null);
+    }
+    return `"${label}"`;
+  }
+
+  public getFieldIndex(id: string) {
+    const dimension = this.dimensionsForSelect().find((d: any) => d.dimension === id);
+    if (dimension) {
+      return super.getFieldIndex(id);
+    }
+    return this.escapeColumnName(this.aliasName(id, false));
+  }
+
+  public seriesSql(timeDimension: any) {
+    const values = timeDimension.timeSeries().map(
+      ([from, to]: [string, string]) => `select '${from}' f, '${to}' t`
+    ).join(' UNION ALL ');
+    return `SELECT ${this.timeStampCast('dates.f')} date_from, ${this.timeStampCast('dates.t')} date_to FROM (${values}) AS dates`;
+  }
+
+  public defaultRefreshKeyRenewalThreshold() {
+    return 120;
```

Comment:
What's this magic number?

Reply:
I'm not sure, it was in the databricks file.
I think the this refresh key has a relation between Cube and front ?